src/chainstate/stacks/transaction.rs

```diff
@@ -113,6 +113,31 @@ impl StacksMessageCodec for TransactionSmartContract {
     }
 }
 
+fn ClarityVersion_consensus_serialize<W: Write>(
+    version: &ClarityVersion,
+    fd: &mut W,
+) -> Result<(), codec_error> {
+    match *version {
+        ClarityVersion::Clarity1 => write_next(fd, &1u8)?,
+        ClarityVersion::Clarity2 => write_next(fd, &2u8)?,
+    }
+    Ok(())
```

Comment:
This can be simplified a bit:

```suggestion
    match version {
        ClarityVersion::Clarity1 => write_next(fd, &1u8),
        ClarityVersion::Clarity2 => write_next(fd, &2u8),
    }
```